src/normalizer/json-normalizer.js

```diff
@@ -39,45 +30,72 @@ function isIgnoredProperty(property, ignoredProperties = DEFAULT_IGNORED_PROPERT
   return !!ignoredProperties[property];
 }
 
-export function normalizeItem(item, picks = null) {
-  const itemTransformation = getTransformation(item);
-  if (!_.isNull(picks)) {
-    // eslint-disable-next-line no-param-reassign
-    item = _.pick(item, _.union(getDefaultJsonItemAttributeNames(), picks));
+function isRelationship(schema, propertyKey, propertyValue) {
+  if (_.get(schema, ['relationships', propertyKey])) {
+    return true;
   }
 
-  return _.reduce(item, (normalizedItem, val, property) => {
-    if (!itemTransformation || isIgnoredProperty(property)) {
+  // TODO: use rio.options to turn off this feature
+  if (
+    _.get(propertyValue, 'id') &&
+    _.get(propertyValue, 'type')
+  ) {
+    return true;
+  }
+
+  return false;
+}
+
+function resolveSchema(item, schema) {
+  const itemSchema = schema || getSchema(item);
+  if (itemSchema) {
+    return itemSchema;
+  }
+  return null;
+}
+
+export function shouldNormalize(item) {
```

Comment:
It would be nice if we could have different normalizers depending on the source type, for example, this should be triggered only for JSON_API source.

Reply:
In ideal situation this would be serialization module (normalize would become serialize). Because we would have in redux state `RIO object model` and denormalization/normalization would only resolve references. While serializers would transform payload back and forth between server and redux state. In that way we would only have `1` type of normalizer/denormalizer and `n` serializers where `n` is number of different source types. 

Something like `rio.registerSources(sourceType, serializer);` and then `rio.normalize(item, schema)`.

But also define serialization in `rio.registerResource` as { serialize, deserialize }.

In that way rio would define serialization as part of registerSource and other part from config.